### PR TITLE
Warning police

### PR DESCRIPTION
**Pull request by graydon**
While diagnosing failures in #3059 I found gcc 9 was emitting a lot of warnings. Decided to mop them up.

### CODE REVIEW

File: src/main/Config.cpp [hunk 1]
```diff
@@ -275,10 +276,24 @@ readInt(ConfigItem const& item, T min = std::numeric_limits<T>::min(),
         throw std::invalid_argument(fmt::format("invalid '{}'", item.first));
     }
     int64_t v = item.second->as<int64_t>()->get();
-    if (v < min || v > max)
+    if (std::is_signed_v<T>)
```
Comment (MonsieurNicolas) [inline src/main/Config.cpp:279]: C++ 17 makes this kind of things so nice

File: src/test/FuzzerImpl.h [hunk 2]
```diff
@@ -22,7 +22,7 @@ struct Operation;
 namespace FuzzUtils
 {
 size_t static constexpr NUM_STORED_LEDGER_KEYS = 0x100U;
-size_t static constexpr NUM_UNVALIDATED_LEDGER_KEYS = 0x40U;
+ssize_t static constexpr NUM_UNVALIDATED_LEDGER_KEYS = 0x40;
```
Comment (MonsieurNicolas) [inline src/test/FuzzerImpl.h:25]: `ssize_t` is not what you want (this is a C standard type not C++). I imagine that you're trying to silence the warning in `generateStoredLedgerKeys`?

If so the type should be `StoredLedgerKeys::iterator::difference_type` (on my machine this is `ptrdiff_t` but I don't know if this is true across platforms)

**Review by MonsieurNicolas — CHANGES_REQUESTED** [review]

File: src/main/Config.cpp (hunk 1 above)
Comment (graydon) [reply src/main/Config.cpp:279]: Quite!

File: src/test/FuzzerImpl.h (hunk 2 above)
Comment (graydon) [reply src/test/FuzzerImpl.h:25]: Fixed (it's just `StoredLedgerKeys::difference_type`)

**Comment by MonsieurNicolas** [comment]
r+ 0a6a8ca55f15945978a5bd710267e442bd181e87

**Pull request merged**